src/EnergyPlus/WeatherManager.cc

```diff
@@ -281,6 +282,11 @@ namespace WeatherManager {
 	Array1D< SpecialDayData > SpecialDays;
 	Array1D< DataPeriodData > DataPeriods;
 
+	std::shared_ptr< BaseGroundTempsModel > siteShallowGroundTempsPtr;
```

Comment:
So the factory method should create the new shared pointer in the ground temp manager, and store it in the vector.  Then anything else that later needs a ground temp model can call the factory method and share ownership of that model instance, and call methods to get ground temperature.  I think that is the proper place for a shared_ptr, right?